lib/protobuf/field/base_field.rb

```diff
@@ -195,6 +195,7 @@ def define_array_setter
 
         message_class.class_eval do
           define_method(method_name) do |val|
+            @memoized_encoded = nil
```

Comment:
I think we might want to say `@has_change = true`. It becomes more clear to me what we're trying to keep track of. That's just my opinion though.